File: java/spark/loadCsv.java

public static class ParseLine implements FlatMapFunction<Tuple2<String, String>, String[]> {
    public Iterable<String[]> call(Tuple2<String, String> file) throws Exception {
        CSVReader reader = new CSVReader(new StringReader(file._2()));
        return reader.readAll(); 
    }
}

JavaPairRDD<String, String> csvData = sc.wholeTextFiles(inputFile); 
JavaRDD<String[]> keyedRDD = csvData.flatMap(new ParseLine());

File: scala/spark/loadCsv.scala
case class Person(name: String, favoriteAnimal: String)

val input = sc.wholeTextFiles(inputFile) 

val result = input.flatMap { case (_, txt) =>
    val reader = new CSVReader(new StringReader(txt));
    reader.readAll().map(x => Person(x(0), x(1))) 
}